libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java

```diff
@@ -844,16 +847,28 @@ public void run() {
         }).start();
     }
 
-    private void postPushUnregister(BroadcastReceiver pushReceiver,
+    private synchronized void postPushUnregister(BroadcastReceiver pushReceiver,
     		final ClientManager clientMgr, final boolean showLoginPage,
     		final String refreshToken, final String loginServer,
             final Account account, Activity frontActivity) {
-        try {
-            context.unregisterReceiver(pushReceiver);
-        } catch (Exception e) {
-            SalesforceSDKLogger.e(TAG, "Exception occurred while un-registering", e);
+
+        /*
+         * This method can be triggered from 2 places - either when the push un-registration call
+         * completes, or when the server times out. Since they're on different threads, we don't
+         * know when either of those calls returns. However, the code should be executed only
+         * once. We also need to make sure that the value is reset once both calls come in,
+         * to ensure that push un-registration works in the multi-user scenario.
+         */
+        final Boolean postPushCalledBool = postPushCalled.get(account);
+        if (postPushCalledBool == null || !postPushCalledBool) {
+            postPushCalled.put(account, true);
```

Comment:
@wmathurin This should work better. This ensures that this code is executed only once, regardless on when/where the call comes from. It also ensures that the multi-user scenario works, since the `boolean` is per `UserAccount`.